src/core/read/py_fread.cc

```diff
@@ -37,20 +37,201 @@ static const char* doc_fread =
+
+skip_blank_lines: bool
+    If `True` then any empty lines in the input will be skipped. If
+    this parameter is `False` then: (a) in single-column mode empty
+    lines are kept as empty lines; otherwise (b) if `fill=True` then
+    empty lines produce a single line filled with NAs in the output;
+    otherwise (c) an :exc:`IOError` is raised.
+
+strip_whitespace: bool
+    If True, then the leading/trailing whitespace will be stripped
+    from unquoted string fields. Whitespace is always skipped from
+    numeric fields.
+
+quotechar: '"' | "'" | "`"
+    The character that was used to quote fields in the CSV file. By
+    default the double-quote mark `'"'` is assumed.
+
+tempdir: str | None
+    Use this directory for storing temporary files as needed. If not
+    provided then the system temporary directory will be used, as
+    determined via the :mod:`tempfile` Python module.
+
+nthreads: int | None
+    Number of threads to use when reading the file. This number cannot
+    exceed the number of threads in the pool ``dt.options.nthreads``.
+    If 0 or negative number of threads is requested, then it will be
+    treated as that many threads less than the maximum. By default
+    all threads in the thread pool are used.
+
+verbose: bool
+    If `True`, then print detailed information about the internal
+    workings of fread to stdout (or to `logger` if provided).
+
+logger: object
+    Logger object that will receive verbose information about fread's
+    progress. When this parameter is specified, `verbose` mode will
+    be turned on automatically.
+
+multiple_sources: "warn" | "error" | "ignore"
+    Action that should be taken when the input resolves to multiple
+    distinct sources. By default (`"warn"`) a warning will be issued
+    and only the first source will be read and returned as a Frame.
+    The `"ignore"` action is similar, except that the extra sources
+    will be discarded without a warning. Lastly, an :exc:`IOError`
+    can be raised if the value of this parameter is `"error"`.
+
+    If you want all sources to be read instead of only the first one
+    then consider using :func:`iread()`.
+
+memory_limit: int
+    Try not to exceed this amount of memory allocation (in bytes)
+    when reading the data. This limit is advisory and not enforced
+    very strictly.
+
+    This setting is useful when reading data from a file that is
+    substantially larger than the amount of RAM available on your
+    machine.
+
+    When this parameter is specified and fread sees that it needs
+    more RAM than the limit in order to read the input file, then
+    it will dump the data that was read so far into a temporary file
+    in binary format. In the end the returned Frame will be partially
+    composed from data located on disk, and partially from the data
+    in memory. It is advised to either store this data as a Jay file
+    or filter and materialize the frame (if not the performance may
```

Comment:
May be to be consistent we should use `:class:Frame` everywhere instead of `Frame` or `frame`? Or at least always say `Frame`.

Reply:
`:class:Frame` generates a link, so might not be appropriate to use it everywhere. Plus it looks too verbose in plain help files.
"Frame" in backticks is the name of the class, but can also be used to talk about an object of this class (when it seems necessary to emphasize the class). 
Lastly, lowercase "frame" is a generic frame/table object.

The distinction between "Frame" and "frame" is kinda fragile, and I'm not sure whether we just need to abandon all uses of "frame", but it seems that in some contexts lowercase is more appropriate...

Reply:
Yeah, but it looks kind of strange that we use three different names for basically the same thing :)